tests/WE2E/run_we2e_tests.py

```diff
@@ -0,0 +1,734 @@
+                "CRON_RELAUNCH_INTVL_MNTS": args.cron_relaunch_intvl_mnts,
+                "EXPT_SUBDIR": test_name,
+                "USE_CRON_TO_RELAUNCH": args.launch == "cron",
+                },
+        }
+
+        if run_envir:
+            test_config_updates["user"].update({"RUN_ENVIR": run_envir})
+
+        workflow = test_config_updates["workflow"]
+        # Adds an item to the dict only if it has a value
+        update = lambda k, v: v and workflow.update({k: v})
+        update("CRON_RELAUNCH_INTVL_MNTS", args.cron_relaunch_intvl_mnts)
+        update("DEBUG", args.debug_tests)
+        update("EXPT_BASEDIR", args.expt_basedir)
+        update("EXEC_SUBDIR", args.exec_subdir)
+        update("VERBOSE", args.verbose_tests)
+
+        test_cfg.update_from(test_config_updates)
+        logging.debug(
+            f"Overwriting WE2E-test-specific settings for test \n{test_name}\n"
+        )
+
+        if "task_get_extrn_ics" in test_cfg:
+            test_cfg["task_get_extrn_ics"] = check_task_get_extrn_bcs(
+                test_cfg, machine_defaults, config_defaults, "ics"
+            )
+        if "task_get_extrn_lbcs" in test_cfg:
+            test_cfg["task_get_extrn_lbcs"] = check_task_get_extrn_bcs(
+                test_cfg, machine_defaults, config_defaults, "lbcs"
+            )
+
+        if "verification" in test_cfg:
+            # This section checks if we are doing verification on a machine with staged verification
+            # obs. If so, and if the config file does not explicitly set the observation locations,
+            # fill these in with defaults from the machine files
+            obs_vars = [
+                "CCPA_OBS_DIR",
+                "MRMS_OBS_DIR",
+                "NDAS_OBS_DIR",
+                "NOHRSC_OBS_DIR",
+            ]
+            for obvar in obs_vars:
+                mach_path = machine_defaults["platform"].get("TEST_" + obvar)
+                if not test_cfg["verification"].get(obvar) and mach_path:
+                    logging.debug(f"Setting {obvar} = {mach_path} from machine file")
+                    test_cfg["verification"][obvar] = mach_path
+
+        if args.compiler == "gnu":
+            # 2D decomposition doesn't work with GNU compilers.  Deactivate 2D decomposition for GNU
+            if "task_run_post" in test_cfg:
+                test_cfg["task_run_post"]["envvars"]["NUMX"] = 1
+                logging.info(
+                    "NUMX has been reset to 1 due to issues encountered with GNU compilers"
+                )
+            if "task_run_fcst" in test_cfg:
+                test_cfg["task_run_fcst"]["ITASKS"] = 1
+                logging.info(
+                    "ITASKS has been reset to 1 due to issues encountered with GNU compilers"
+                )
+
+        logging.debug(
+            f"Writing updated config.yaml for test {test_name}\n"
+            "based on specified command-line arguments:\n"
+        )
+        logging.debug(print(test_cfg))
```

Comment:
This print command is writing to screen regardless of specified logging level. I believe this is the correct way to do what you want here.
```suggestion
        logging.debug(str(test_cfg))
```